webui/react/src/components/Metadata.tsx

```diff
@@ -0,0 +1,95 @@
+import Button from 'hew/Button';
+import Icon from 'hew/Icon';
+import Message from 'hew/Message';
+import Surface from 'hew/Surface';
+import { useTheme } from 'hew/Theme';
+import Tooltip from 'hew/Tooltip';
+import Tree, { TreeDataNode } from 'hew/Tree';
+import { isArray } from 'lodash';
+
+import { JsonObject, TrialDetails } from 'types';
+import { downloadText } from 'utils/browser';
+import { isJsonObject } from 'utils/data';
+
+import css from './Metadata.module.scss';
+import Section from './Section';
+
+interface Props {
+  trial?: TrialDetails;
+}
+
+export const EMPTY_MESSAGE = 'No metadata found';
+
+const Metadata: React.FC<Props> = ({ trial }: Props) => {
+  const { tokens } = useTheme();
+
+  const getNodes = (data: JsonObject): TreeDataNode[] => {
+    return Object.entries(data).map(([key, value]) => {
+      if (isJsonObject(value)) {
+        return {
+          children: getNodes(value),
+          key,
+          selectable: false,
+          title: <span style={{ color: tokens.colorTextDescription }}>{key}</span>,
+        };
+      } else if (isArray(value)) {
+        return {
+          children: getNodes(JSON.parse(JSON.stringify(Object.assign({}, value)))),
```

Comment:
```suggestion
          children: { ...value },
```
?

Reply:
realized a lot of this could be simplified if I just used the right JSON type 🤦 updated